File: solutions/typescript/0013.ts
function romanToInt(s: string): number {
    const romanMap = new Map<string, number>([
        ['I', 1],
        ['V', 5],
        ['X', 10],
        ['L', 50],
        ['C', 100],
        ['D', 500],
        ['M', 1000]
    ]);
    let result = 0;
    for (let i = 0; i < s.length; i++) {
        const current = romanMap.get(s[i]);
        const next = romanMap.get(s[i + 1]);
        
        if (!current) {
            throw new Error(`无效的罗马数字字符: ${s[i]}`);
        }
        
        if (next && current < next) {
            result -= current;
        } else {
            result += current;
        }
    }
    return result;
};

File: solutions/go/0013.go
func romanToInt(s string) int {
	romanMap := map[byte]int{
		'I': 1,
		'V': 5,
		'X': 10,
		'L': 50,
		'C': 100,
		'D': 500,
		'M': 1000,
	}
	
	result := 0
	for i := 0; i < len(s); i++ {
		current := romanMap[s[i]]
		
		// 检查是否有下一个字符
		if i+1 < len(s) {
			next := romanMap[s[i+1]]
			if current < next {
				result -= current
			} else {
				result += current
			}
		} else {
			result += current
		}
	}
	
	return result
}